test/function/entry.cpp

```diff
@@ -97,6 +98,76 @@ namespace {
       }
   };
 
+  TEST_F( EntryTest, EntryToStringMallocFailure ) {
+    void * (*set_malloc_result)(size_t);
+    const char *result;
+
+    set_malloc_result = stumpless_set_malloc( MALLOC_FAIL );
+    ASSERT_NOT_NULL( set_malloc_result );
+
+    result = stumpless_entry_to_string(basic_entry);
+    EXPECT_ERROR_ID_EQ( STUMPLESS_MEMORY_ALLOCATION_FAILURE );
+    EXPECT_NULL( result );
+
+    set_malloc_result = stumpless_set_malloc( malloc );
+    EXPECT_TRUE( set_malloc_result == malloc );
+  }
+
+  TEST_F(EntryTest, OutputFormatCheck) {
+    char *result = stumpless_entry_to_string(basic_entry);
+    ASSERT_NE(result, nullptr);
+
+    const char *expected_output =
+      "prival=\"14\",app_name=\"basic-app-name\",hostname=\"\","
+      "message_id=\"basic-msgid\",process_id=\"\","
+      "message=\"basic message\","
+      "basic-element=[basic-param=\"basic-value\"],"
+      "basic-element-2";
+    EXPECT_STREQ(result, expected_output);
+    size_t len = strlen(result);
+    EXPECT_NE(result[len - 1], ',');
+    free(result);
+  }
+
+  TEST_F(EntryTest, LastCommaCheck) {
+    char *result = stumpless_entry_to_string(basic_entry);
+    ASSERT_NE(result, nullptr);
+
+    size_t len = strlen(result);
+    EXPECT_NE(result[len - 1], ',');
+
+    free(result);
+  }
+
+  TEST_F(EntryTest, NullEntry) {
+    char *result = stumpless_entry_to_string(NULL);
+    EXPECT_EQ(result, nullptr);
+    free(result);
+  }
+
+  TEST_F(EntryTest, EmptyHostnameAndProcid) {
+    char *result = stumpless_entry_to_string(basic_entry);
+    ASSERT_NE(result, nullptr);
+    EXPECT_TRUE(strstr(result, "hostname=\"\"") != nullptr);
+    EXPECT_TRUE(strstr(result, "process_id=\"\"") != nullptr);
+    EXPECT_TRUE(strstr(result, "message_id=\"basic-msgid\"") != nullptr);
+    free(result);
+  }
+
+  TEST_F(EntryTest, NoneOfElementCheck) {
+    struct stumpless_entry test_entry = *basic_entry;
```

Comment:
Instead of `basic_entry`, use `create_empty_entry()`, which will create an entry without any elements. You could either add this to the fixture setup/teardown, or just call the function in this test. This will avoid depending on struct copy mechanics and internal fields of the entry struct. The `nil_entry` fixture is also an option for this test I believe, which is already handled by the fixture setup.